src/fastlock.h

```diff
@@ -6,10 +6,16 @@
 extern "C" {
 #endif
 
+typedef int (*spin_worker)();
+
 /* Begin C API */
 struct fastlock;
 void fastlock_init(struct fastlock *lock, const char *name);
-void fastlock_lock(struct fastlock *lock);
+#ifdef __cplusplus
+void fastlock_lock(struct fastlock *lock, spin_worker worker = nullptr);
+#else
+void fastlock_lock(struct fastlock *lock, spin_worker worker);
+#endif
```

Comment:
So fastlock_lock needs a worker argument if compiled in C? I know it won't ever come up realistically but it seems strange to have this guard here when it won't compile anyway.

Reply:
fastlock has a C api, and there's no reason you can't call this from C code with a simple C function.